include/internal/constant_time_locl.h

```diff
@@ -213,18 +213,66 @@ static ossl_inline unsigned char constant_time_eq_int_8(int a, int b)
     return constant_time_eq_8((unsigned)(a), (unsigned)(b));
 }
 
+/* Returns the value unmodified, but avoids optimizations. */
+static ossl_inline unsigned int value_barrier(unsigned int a)
+{
+#if !defined(OPENSSL_NO_ASM) && defined(__GNUC__)
+    unsigned int r;
+    __asm__("" : "=r"(r) : "0"(a));
```

Comment:
I don't understand how this is supposed to work. We're going to write to `r`, and have an input `a`, but we don't have any instructions that actually do anything with any of them. Are you just using some implementation detail that happens to put them in the same register? Is there a reason you don't use the same solution as boringssl uses that just say we read and write `a`?

Reply:
By the "0" I place a into the same register that will hold the result.
GCC does not make assumptions what an empty assembler statement can do.

Reply:
"0" is an index in the output section.
I could swap this way:
asm("", "=r"(r), "=r(s)" : "0"(s) , "1"(r"))

Reply:
I would prefer the variant with the volatile variable, but it forces r to be in memory,
while boring ssl does not handle non-gcc and/or no-asm.